src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\Debug\ExceptionHandler;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * HttpExceptionListener.
+ *
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+          KernelEvents::EXCEPTION => array(
+            array('onKernelException', -256),
+          ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if ($event->hasResponse()) {
+            return;
+        }
+
+        $flattenException = FlattenException::create($event->getException());
+        $handler = new ExceptionHandler(false);
+        $response = new Response($handler->getContent($flattenException));
```

Comment:
@nicolas-grekas but now we are still not using any headers set on the http exception.

Should we really use `ExceptionHandler` here? Just for the content of the response?

Imho the content like "404 not found" is totally fine?

https://github.com/symfony/symfony/pull/26928/files#diff-3d3b8968ae2d94a9aab7078dec41e607R228

Reply:
I think we should yes, it matches what adding the global handlers does.
The output is nicer and we have a central place to improve it if needed.
In dev, this could also be useful (but the debug flag should be properly injected of course)

Reply:
Ok i see that point :+1: only header handling missing here then :blush: